### PR TITLE
Ablation

### PR DESCRIPTION
**Pull request by a1302z**

### CODE REVIEW

File: revision_scripts/module_modification.py [hunk 1]
```diff
@@ -88,6 +88,26 @@ def matchDim():
     return matchDim()(module.num_features)
 
 
+def _batchnorm_to_bn_without_stats(
+    module: nn.modules.batchnorm._BatchNorm, keep_running_stats: bool = True
```
Comment (gkaissis) [inline revision_scripts/module_modification.py]: why is `keep_running_stats` here True?

File: syft/frameworks/torch/fl/dataloader.py [hunk 2]
```diff
@@ -209,17 +256,26 @@ def __init__(
...
         self.collate_fn = collate_fn
-        self.iter_class = _DataLoaderOneWorkerIter if iter_per_worker else _DataLoaderIter
+        self.iter_class = (
+            _DataLoaderOneWorkerIter if iter_per_worker else _DataLoaderIter
+        )
 
         # Build a batch sampler per worker
         self.batch_samplers = {}
         for worker in self.workers:
+            """Behold the case for a switch-case statement in Python."""
```
Comment (gkaissis) [inline syft/frameworks/torch/fl/dataloader.py:266]: the new match/case construct in python 3.10 is going to be very nice for this

File: test.py [hunk 3]
```diff
@@ -83,6 +110,37 @@
...
+        basic_tfs = [
+            a.Resize(args.inference_resolution, args.inference_resolution,),
+            a.RandomCrop(args.train_resolution, args.train_resolution),
+            a.ToFloat(max_value=255.0),
+        ]
+        val_trans = a.Compose(
+            [
+                *basic_tfs,
+                a.Normalize(mean, std, max_pixel_value=1.0),
+                a.Lambda(
```
Comment (gkaissis) [inline test.py:123]: is this where "lambda is incompatible with multiprocessing" comes from?

File: test.py [hunk 4]
```diff
@@ -161,17 +270,127 @@
...
+        interesting_idcs = np.concatenate([interesting_idcs[:5], interesting_idcs[-5:]])
+
+        plot_imgs(
+            imgs[interesting_idcs],
+            total_target[interesting_idcs],
+            total_pred[interesting_idcs],
+            savefig="test.png",
+        )
+
+        # plt.clf()
```
Comment (gkaissis) [inline test.py:362]: Should we remove the unused comments here? @NiWaRe

File: torchlib/DPUtility_ablation.py [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+from numpy import generic as np_generic
+from pathlib import Path
+from itertools import product
+
+sys.path.insert(0, os.path.split(sys.path[0])[0])
+
+from train import main
+
+
+def ablation(args, noise_multiplier, max_grad_norm):
```
Comment (gkaissis) [inline torchlib/DPUtility_ablation.py:15]: same here, should we remove the comments? @NiWaRe

**Review by gkaissis — COMMENTED** [review]
Left some comments, looks good overall. I can't check 100% of it since I wasn't involved in all the actual coding. Feel free to merge otherwise!

File: test.py (hunk 3 above)
Comment (NiWaRe) [reply test.py:123]: Yes there're a couple of lambdas in the data-loading, I think I created an issue for that (if not I'll create one).

File: test.py (hunk 4 above)
Comment (NiWaRe) [reply test.py:362]: Yes sure there're actually a ton of comments I still need to remove, will do so asap.

File: torchlib/DPUtility_ablation.py (hunk 5 above)
Comment (NiWaRe) [reply torchlib/DPUtility_ablation.py:15]: Yes, same

Comment (NiWaRe) [reply torchlib/DPUtility_ablation.py:15]: Except for one in the federated data loading section, some of them I added deliberately - to be uncommented once #8 is resolved.

File: revision_scripts/module_modification.py (hunk 1 above)
Comment (a1302z) [reply revision_scripts/module_modification.py]: it was not used, but fixed in 1aa7f769d827df1313f247349b30fd6d3658cc18

**Pull request merged**